src/components/__tests__/DynamicWidgets.js

```diff
@@ -1,89 +1,76 @@
-import { mount } from '../../../test/utils';
+import { mount, nextTick } from '../../../test/utils';
 import DynamicWidgets from '../DynamicWidgets';
 import { __setState } from '../../mixins/widget';
 import { AisPanel } from '../../widgets';
 jest.mock('../../mixins/widget');
 
 const MockRefinementList = {
   props: { attribute: { type: String } },
-  render(h) {
-    return h('div', {
-      attrs: {
-        'widget-name': 'ais-refinement-list',
-        attribute: this.attribute,
-      },
-    });
-  },
+  template: `
+    <div widget-name="ais-refinement-list">{{ attribute }}</div>
```

Comment:
In vue 3, `attribute: this.attribute` was not rendered on `div` properly. So I've moved it as textContent.

Reply:
good enough, but I wonder if something like `attribute-name` could work

Reply:
You're right. I didn't try `:attribute="attribute"` after changing it to template form. Now that I tried it, it just works. Now the snapshots are the same as before. https://github.com/algolia/vue-instantsearch/pull/1021/commits/9676a4621cdce5315874cf80b0eb742149a05594

Reply:
the test on vue 3 fails now though, is there some special case for props that don't get added to elements that don't "support" it? maybe data-attribute worst case? Also putting it all in text is fine I think (maybe json stringified)

Reply:
Ah okay, I've got a better understanding.
https://codesandbox.io/s/busy-roentgen-3opm3?file=/src/components/HelloWorld.vue:13-172

`attributes` as an attribute is not rendered by Vue 3 at all.

Reply:
<img width="309" alt="Screenshot 2021-07-20 at 11 34 57" src="https://user-images.githubusercontent.com/6270048/126299227-375a8644-9457-4e07-ad2b-8315040002b1.png">

`attributes` specifically is special cased it seems?

Reply:
https://user-images.githubusercontent.com/499898/126318350-c4ff30a8-803c-4f8a-909b-487f12c8d0fe.mp4

oh I didn't know about this ↑

Reply:
let's use `data-attribute` then :)

Reply:
yeah, updated it to use JSON.stringify() https://github.com/algolia/vue-instantsearch/commit/f73aefa1b1b761178936f7eda653022eae337818